src/mixer.h

```diff
@@ -56,51 +56,29 @@ class Mixer : public cyclus::Facility {
   virtual void InitInv(cyclus::Inventories& inv);
 
  protected:
-  #pragma cyclus var { \
-    "doc": "Ordered list of commodities on which to request material stream" \
-    " material, each commodity corresponds to an input stream.", \
-    "uilabel": "Input Commodities", \
-    "uitype": ["oneormore", "incommodity"], \
+#pragma cyclus var { \
+    "alias": ["in_streams", [ "stream", [ "info", "mixing_ratio", "buf_size"], [ "commodities", "commodity", "pref"]]], \
+    "uitype": ["oneormore", [ "pair", ["pair", "double", "double"], ["oneormore", "incommodity", "double"]]], \
+    "uilabel": "", \
+    "doc": "", \
   }
-  std::vector<std::string> in_commods;
-
-  #pragma cyclus var { \
-    "default": [], \
-    "uilabel": "Input Preferences", \
-    "doc": "Stream commodity request preferences for each of the given" \
-           "commodities (same order)." \
-           " If unspecified, default is 1.0 for all preferences.", \
-  }
-  std::vector<double> in_commod_prefs;
+  std::vector<std::pair<std::pair<double, double>, std::map<std::string, double> > > streams_;
 
-  #pragma cyclus var { \
-    "doc": "Size of input material stream inventory - i.e. the quantity of each" \
-           " stream type to keep on-hand)", \
-    "uilabel": "Input Inventory Capacity", \
-    "units": [ "", "kg"],                  \
-  }
+  std::vector<std::map<std::string, double> > in_commods;
```

Comment:
`streams_` contains the full information and are implemented in cyclus input, `in_commods`, `in_buf_sizes` and `mixing_ratios` are remaining from the previous API, allowing a easy switch, without a complete implementation change...
I just have to fill `in_commods`, `in_buf_sizes` and `mixing_ratios` with the streams_ information provided by the users to make my stuff working...

I am not sure I want to remove it, because it also allow dealing with simple type instead of playing all the time with a `vector<pair<double,double>,map<string,double>>>`